src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -580,162 +671,185 @@ public TypeSymbolWithAnnotations SetUnknownNullabilityForReferenceTypes()
         {
             throw ExceptionUtilities.Unreachable;
         }
 
-        [Obsolete("Unsupported", error: true)]
-        public static bool operator ==(TypeSymbolWithAnnotations x, TypeSymbolWithAnnotations y)
+        // Field-wise ReferenceEquals.
+        public static bool operator ==(TypeSymbolWithAnnotations? x, TypeSymbolWithAnnotations? y)
         {
-            throw ExceptionUtilities.Unreachable;
+            return AreFieldsReferenceEquals(x.GetValueOrDefault(), y.GetValueOrDefault());
         }
 
-        [Obsolete("Unsupported", error: true)]
-        public static bool operator !=(TypeSymbolWithAnnotations x, TypeSymbolWithAnnotations y)
+        // Field-wise ReferenceEquals.
+        public static bool operator !=(TypeSymbolWithAnnotations? x, TypeSymbolWithAnnotations? y)
         {
-            throw ExceptionUtilities.Unreachable;
+            return !(x == y);
         }
 
-        [Obsolete("Unsupported", error: true)]
-        public static bool operator ==(Symbol x, TypeSymbolWithAnnotations y)
+        private static bool AreFieldsReferenceEquals(TypeSymbolWithAnnotations x, TypeSymbolWithAnnotations y)
         {
-            throw ExceptionUtilities.Unreachable;
+            return ReferenceEquals(x._defaultType, y._defaultType) &&
+                ReferenceEquals(x.NonNullTypesContext, y.NonNullTypesContext) &&
+                x.IsAnnotated == y.IsAnnotated &&
+                ReferenceEquals(x._extensions, y._extensions);
         }
 
-        [Obsolete("Unsupported", error: true)]
-        public static bool operator !=(Symbol x, TypeSymbolWithAnnotations y)
+        /// <summary>
+        /// Additional data or behavior beyond the core TypeSymbolWithAnnotations.
+        /// </summary>
+        private abstract class Extensions
         {
-            throw ExceptionUtilities.Unreachable;
-        }
+            internal static readonly Extensions Default = new NonLazyType(ImmutableArray<CustomModifier>.Empty);
 
-        [Obsolete("Unsupported", error: true)]
-        public static bool operator ==(TypeSymbolWithAnnotations x, Symbol y)
-        {
-            throw ExceptionUtilities.Unreachable;
-        }
+            internal static Extensions Create(ImmutableArray<CustomModifier> customModifiers)
+            {
+                if (customModifiers.IsEmpty)
+                {
+                    return Default;
+                }
+                return new NonLazyType(customModifiers);
+            }
 
-        [Obsolete("Unsupported", error: true)]
-        public static bool operator !=(TypeSymbolWithAnnotations x, Symbol y)
-        {
-            throw ExceptionUtilities.Unreachable;
+            internal static Extensions CreateLazy(CSharpCompilation compilation, TypeSymbolWithAnnotations underlying)
+            {
+                return new LazyNullableTypeParameter(compilation, underlying);
+            }
+
+            public abstract TypeSymbol GetResolvedType(TypeSymbol defaultType);
```

Comment:
>public [](start = 12, length = 6)

nit: there's a mix of `internal` and `public`. Consider standardizing on one (my preference is `internal`, but that's up to you).